presto-main/src/main/java/com/facebook/presto/FullConnectorSession.java

```diff
@@ -31,20 +34,42 @@
     private final Locale locale;
     private final long startTime;
     private final Map<String, String> properties;
+    private final String catalog;
+    private final SessionPropertyManager sessionPropertyManager;
+
+    public FullConnectorSession(
+            String user,
+            TimeZoneKey timeZoneKey,
+            Locale locale,
+            long startTime)
+    {
+        this.user = requireNonNull(user, "user is null");
+        this.timeZoneKey = requireNonNull(timeZoneKey, "timeZoneKey is null");
+        this.locale = requireNonNull(locale, "locale is null");
+        this.startTime = startTime;
+
+        this.properties = null;
```

Comment:
This seems strange. Would it make more sense to have a subclass with the extra properties and method?